File: javascript/ql/src/Expressions/examples/CompareIdenticalValues.js
function Rectangle(x, y, width, height) {
  this.x = x;
  this.y = y;
  this.width = width;
  this.height = height;
}

Rectangle.prototype.contains = function(x, y) {
  return (this.x <= x &&
          x < this.x+this.width) &&
         (y <= y &&
          y < this.y+this.height);
};

File: go/ql/test/query-tests/RedundantCode/CompareIdenticalValues/CompareIdenticalValues.go
package main

type Rectangle struct {
	x, y, width, height int
}

func (r *Rectangle) containsBad(x, y int) bool {
	return r.x <= x &&
		y <= y && // NOT OK
		x <= r.x+r.width &&
		y <= r.y+r.height
}
